File: Remove Duplicates from Sorted Array II.java
public class Solution {
    public int removeDuplicates(int[] A) {
        // Start typing your Java solution below
        // DO NOT write main() function
        if(A.length == 0)
        {
            return 0;
        }
        int count = 0;
        int pos = 0;
        int preVal = A[0]-1;
        for(int i = 0; i < A.length; i++)
        {
            if(A[i]==preVal)
            {
                count++;
            }
            if(A[i]!=preVal || count<2)
            {
                A[pos] =  A[i];
                pos++;
                if(A[i]!=preVal)
                {
                    count = 0;   
                }
            }
            preVal = A[i];
        }
        
        return pos;
        
    }
}

File: Remove Duplicates from Sorted Array II.cpp
class Solution {
public:
    int removeDuplicates(int A[], int n) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        if(n==0)
            return 0;
        
        int count = 0;
        int cur = 0;
        for(int i = 1; i < n; i++)
        {
            if(A[i-1]!=A[i])
            {
                cur++;
                A[cur] = A[i];
                count = 0;
            }
            else
            {
                if(count==0)
                {
                    cur++;
                    A[cur] = A[i];
                    count++;
                }
            }
        }
        
        return cur+1;
    }
};